source/checkpoint_restart.f90

```diff
@@ -866,7 +863,7 @@ subroutine crpoint
     backspace(6,iostat=ierro)
     rewind(lout)
     endfile(lout,iostat=ierro)
-    close(lout)
+    call f_close(lout)
```

Comment:
I don't know. Not my code.

Reply:
It is; after copying whatever is in lout into fort.6 (as one does when CR-ing), it flushes fort.6 (endfile/backspace pattern), then goes completely overkill to truncate the file with rewind/endfile/close/reopen... Then there is an "else" staement with if(sixrecs /= osixrecs)... This code should probably reviewed and at least documented with a few comments at some point, because it all seems a bit weird.